### ISSUE TITLE
[BUG] Stop button does not work on Intense Packet Generator tool

### ISSUE THREAD

**Issue by enski6** [dannagle/PacketSender#320]
#### What OS and Version?
Windows 10 
PacketSender version 8.3.6
Two computers hooked directly together with 10Gb fiber.
#### Description of issue
While using the Intense Packet Generator tool, setting the speed to 1250 MB/sec the Stop button does not work.  In order to stop sending packets I have to close the application.

I am also only able to send packets at a max rate of about 230,000 packets/sec, how can one send at a higher rate?

Thanks

**Comment by dannagle** [comment]
I can't recreate start/stop GUI problem on my end.  I don't have 10G fiber though. Perhaps that level is what messes with the event loop.

There is a command line version that is able to run faster:
https://github.com/dannagle/PacketSender#cliintensetraffic

You can give that one a try.

**Comment by enski6** [comment]
Thanks, in using the CLI version I am only able to get around 90,000 packets/sec by setting the rate to 0.  If I use the bps setting the most I could generate was about 1000 packets/sec.  
packetsender --rate 0  --num 10000000 -u 192.168.2.100 12345 "hello"

**Comment by dannagle** [comment]
That is odd it is actually slower. I will have to look into that. Can you launch multiple terminal windows with multiple instances running?

**Comment by enski6** [comment]
Yes, I can open another instance and run them simultaneously.  2 instances produced about 150,000 packets/sec total based on Wireshark(packets/time).

**Comment by dannagle** [comment]
That's good. That tells me I can solve this problem by creating additional threads to help with the sending. 

Does it matter if the incoming data is coming from multiple source ports?  I can't use the same socket in multiple threads (well, not easily).

**Comment by enski6** [comment]
Multiple ports are fine.  We are just trying to max out packets/sec using small packets to simulate server load.

**Comment by dannagle** [comment]
If you don't mind, will you email me at hello@packetsender.com ? I plan to create an alpha build for you to try out.
